server/server/src/file_service.rs

```diff
@@ -202,6 +221,23 @@ pub async fn change_doc(
         )?
         .to_lazy();
 
+        let old_usage = get_usage(&tree, db.sizes.data(), None)
```

Comment:
you probably don't need to answer the question in this complicated way here, you can just determine whether the new file is smaller than the older file (if they're over the cap)

Reply:
if i do that, the file metadata cost would be disregarded.